ibis-server/app/model/data_source.py

```diff
@@ -24,31 +26,60 @@ def get_connection(self, dto):
                 raise NotImplementedError(f'Unsupported data source: {self}')
 
     @staticmethod
-    def get_postgres_connection(dto: PostgresDTO):
+    def get_postgres_connection(info: 'PostgresConnectionInfo') -> BaseBackend:
         return ibis.postgres.connect(
-            user=dto.user,
-            password=dto.password,
-            database=dto.database,
-            host=dto.host,
-            port=dto.port,
+            user=info.user,
+            password=info.password,
+            database=info.database,
+            host=info.host,
+            port=info.port,
         )
 
     @staticmethod
-    def get_bigquery_connection(dto: BigQueryDTO):
-        credits_json = loads(base64.b64decode(dto.credentials).decode('utf-8'))
+    def get_bigquery_connection(info: 'BigQueryConnectionInfo') -> BaseBackend:
+        credits_json = loads(base64.b64decode(info.credentials).decode('utf-8'))
         credentials = service_account.Credentials.from_service_account_info(credits_json)
         return ibis.bigquery.connect(
-            project_id=dto.project_id,
-            dataset_id=dto.dataset_id,
+            project_id=info.project_id,
+            dataset_id=info.dataset_id,
             credentials=credentials,
         )
 
     @staticmethod
-    def get_snowflake_connection(dto: SnowflakeDTO):
+    def get_snowflake_connection(info: 'SnowflakeConnectionInfo') -> BaseBackend:
         return ibis.snowflake.connect(
-            user=dto.user,
-            password=dto.password,
-            account=dto.account,
-            database=dto.database,
-            schema=dto.sf_schema,
+            user=info.user,
+            password=info.password,
+            account=info.account,
+            database=info.database,
+            schema=info.sf_schema,
         )
+
+
+class PostgresConnectionInfo(BaseModel):
+    host: str = Field(examples=["localhost"])
```

Comment:
I think we should also support to use [the connection string of postgres](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING-KEYWORD-VALUE). I found ibis-postgres has [another connection](https://ibis-project.org/backends/postgresql#ibis.connect-url-format) mode.

Because the user may have additional config in the connection(e.g. ssl-enable), it's not enough if we only provide host, port, database, user and password.